userspace/libscap/engine/source_plugin/source_plugin.c

```diff
@@ -190,23 +190,21 @@ static int32_t next(struct scap_engine_handle engine, OUT scap_evt** pevent, OUT
 	// Sanity checks in case a plugin implements a non-syscall event source.
 	// If a plugin has event sourcing capability and has a specific ID, then
 	// it is allowed to proce only plugin events of its own event source.
+	uint32_t* plugin_id = (uint32_t*)((uint8_t*) evt + sizeof(scap_evt) + 4 + 4);
 	if (handle->m_input_plugin->id != 0)
 	{
 		/*
 		* | scap_evt | len_id (4B) | len_pl (4B) | id | payload |
 		* Note: we need to use 4B for len_id too because the
 		* PPME_PLUGINEVENT_E has EF_LARGE_PAYLOAD flag!
 		*/
-		if (evt->type != PPME_PLUGINEVENT_E
-			|| evt->tid != (uint64_t) -1
-			|| evt->nparams != 2)
+		if (evt->type != PPME_PLUGINEVENT_E || evt->nparams != 2)
 		{
 			snprintf(lasterr, SCAP_LASTERR_SIZE, "malformed plugin event produced by plugin: '%s'", handle->m_input_plugin->name);
 			return SCAP_FAILURE;
 		}
 
 		// forcely setting plugin ID with the one of the open plugin
-		uint32_t* plugin_id = (uint32_t*)((uint8_t*) evt + sizeof(scap_evt) + 4 + 4);
 		if (*plugin_id == 0)
```

Comment:
If a plugin has event sourcing capability with ID=0, thus not implementing a specific event source, then it can produce any sort of events and it's trusted to do the right thing. This check simply enforces the fact that `PPME_PLUGINEVENT_E` cannot have ID=0, otherwise we would generate broken events potentially.

Reply:
Ok so it's just a defensive check, thank you!